src/App.tsx

```diff
@@ -1,26 +1,76 @@
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hook';
+import { actions as todosActions } from './features/todos';
+
+export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const { status, query } = useAppSelector(state => state.filter);
+
+  const [loader, setLoader] = useState(false);
+
+  useEffect(() => {
+    setLoader(true);
+
+    getTodos()
+      .then((todo: Todo[]) => {
+        dispatch(todosActions.addTodos(todo));
+      })
+      .finally(() => setLoader(false));
```

Comment:
Consider adding error handling in the promise chain of `getTodos` to manage potential errors during the API call. This can be done by adding a `.catch()` method to handle any errors that occur.